test/metabase/api/pivots.clj

```diff
@@ -5,7 +5,7 @@
 (defn applicable-drivers
   "Drivers that these pivot table tests should run on"
   []
-  (disj (mt/normal-drivers-with-feature :expressions :left-join)
+  (disj (mt/normal-drivers-with-feature :expressions :left-join :foreign-keys)
```

Comment:
Is this too general?

Reply:
There are some drivers that don't actually have foreign keys but should still be able to run these tests. `:bigquery-cloud-sdk` comes to mind. So I don't think we want to exclude everything that doesn't have `:foreign-keys`

If you can't make it work for MongoDB then maybe defining a new `:test/pivot-tests` "feature" that defaults to true or something and having `:mongo` and `:redshift` return false might be the way to go here

Reply:
@camsaul  The problem is, these queries trigger the `:foreign-keys` feature check in the `add-implicit-joins` middleware. `:mongo` doesn't support foreign keys, `:bigquery-cloud-sdk` apparently does.

Reply:
It's not every test that uses `pivots/applicable-drivers` that checks for `:foreign-keys`, right? Just some of them? If it's just a few we should probably do the filtering there instead 

Do you know which test it was specifically that was failing? Sort of a weird situation but :bigquery-cloud-sdk` `supports?` returns `false` for `:foreign-keys` when running tests, but we do have a hacky macro called `with-bigquery-fks` to add FK mappings to BigQuery that we use for a few tests. That might be what's happening in the places you saw test failures for Mongo. 

I can rework that macro to turn on the `:foreign-keys` flag so you keep the check and the test(s) would still run for BigQuery but not Mongo

Reply:
1 out of 12 fails for Mongo. `metabase.query-processor.pivot-test/generate-queries-test` is the only one that passes. It doesn't use the functions from `metabase.api.pivots`.

Reply:
I really don't want to start skipping this for BigQuery, I guess maybe we should just come up with a different approach to make sure BigQuery runs `:foreign-key` tests if there's no way to get this PR in without disabling those tests.